### PR TITLE
Added utility code for parsing registration

### PR DESCRIPTION
**Pull request by Mohamed-Abo-El-Soud**
Wasn't expecting this. I know this is a Python repo, but I didn't wanna take too long to learn how to use Python to parse and create a json file so this was way easier.

No unit testing provided. And also a lot of assumptions made in the code about category names. There is a responsibility next year to modify this to fit the use-case. Maybe we should let it recognize category names. Oh yeah, apparently next year this will be taken care of by MyMist so no one will care about this anymore.

@MustafaHaddara review this plz

### CODE REVIEW

File: parse_registration.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    if (compName && compName != "NULL") {
+                        if (!studentIDsGroupedByComp[compName]) {
+                            studentIDsGroupedByComp[compName] = [];
+                        }
+                        if (groupComps.has(compName)) {
+                            if (studentIDsGroupedByComp[compName].indexOf(data.team_id) === -1) {
+                                studentIDsGroupedByComp[compName].push(data.team_id);
+                            }
+                        } else {
+                            studentIDsGroupedByComp[compName].push(data.team_id+"-"+data.mist_id);
```
Comment (MustafaHaddara) [inline parse_registration.js]: did you check with the comps team to confirm that they're including team id on the attendance sheets that judges see? if not, this could be just the mist_id

File: parse_registration.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    "Social Media",
+    "Short Film",
+    "Business Venture",
+    "Humanitarian Service",
+    "Science Fair",
+    "Basketball - Male",
+    "Basketball - Female"
+]);
+
+const studentIDsGroupedByComp = competitions.reduce((acc,comp)=>{acc[comp] = []; return acc},{});
```
Comment (MustafaHaddara) [inline parse_registration.js]: we don't need this `reduce` call since you've got the `if (!studentIDsGroupedByComp[compName])` check farther down?

Comment (Mohamed-Abo-El-Soud) [reply parse_registration.js]: good eye

actually, I want to get rid of the `if` statement eventually. the idea is that I don't like having categories unless i "declare" them. So basically, if I'm expecting `"Mist Quiz Bowl"` not `"Mist Bowl"` . If I let it slide, then I'll have two categories of the same competition but differently spelled. So In the beginning i just made sure the code would throw errors telling you not to do that.

Obviously this backfired when I realized that there are too spelling mistakes (so like `"Nasheed/Rap - Male"` and `"Nasheed /Rap - Male"`, notice the space). So i ended up doing some things manually. But maybe next year ill have a tighter lockdown on the excel sheet.

tldr; i wanna get rid of the `if` statement because its shit code.
